tensorflow_datasets/structured/movielens.py

```diff
@@ -0,0 +1,435 @@
+          format_version='20m',
+          table_option='ratings',
+          download_url=(
+              'http://files.grouplens.org/datasets/movielens/'
+              'ml-20m.zip'
+          ),
+          parsing_fn=movielens_parsing.parse_current_ratings_data,
+      ),
+      MovieLensConfig(
+          name='20m-movies',
+          description=textwrap.dedent("""\
+              This dataset contains data of 27,278 movies rated in the 20m
+              dataset"""),
+          version='0.1.0',
+          format_version='20m',
+          table_option='movies',
+          download_url=(
+              'http://files.grouplens.org/datasets/movielens/'
+              'ml-20m.zip'
+          ),
+          parsing_fn=movielens_parsing.parse_current_movies_data,
+      ),
+  ]
+
+  VERSION = tfds.core.Version('0.1.0')
+
+  def _info(self) -> tfds.core.DatasetInfo:
+    """Returns DatasetInfo according to self.builder_config."""
+    movie_features_dict = {
+        'movie_id': tf.string,
+        'movie_title': tf.string,
+        'movie_genres': tfds.features.Sequence(
+            tfds.features.ClassLabel(names=[
+                'Action', 'Adventure', 'Animation', 'Children', 'Comedy',
+                'Crime', 'Documentary', 'Drama', 'Fantasy', 'Film-Noir',
+                'Horror', 'IMAX', 'Musical', 'Mystery', 'Romance', 'Sci-Fi',
+                'Thriller', 'Unknown', 'War', 'Western', '(no genres listed)',
+            ]),
+        ),
+    }
+    rating_features_dict = {
+        'user_id': tf.string,
+        'user_rating': tf.float32,
+        # Using int64 since tfds currently does not support float64.
+        'timestamp': tf.int64,
+    }
+    demographic_features_dict = {
+        'user_gender': tf.bool,
+        'bucketized_user_age': tf.float32,
+        'user_occupation_label': tfds.features.ClassLabel(names=[
+            'academic/educator', 'artist', 'clerical/admin', 'customer service',
+            'doctor/health care', 'entertainment', 'executive/managerial',
+            'farmer', 'homemaker', 'lawyer', 'librarian', 'other/not specified',
+            'programmer', 'retired', 'sales/marketing', 'scientist',
+            'self-employed', 'student', 'technician/engineer',
+            'tradesman/craftsman', 'unemployed', 'writer',
+        ]),
+        'user_occupation_text': tf.string,
+        'user_zip_code': tf.string,
+    }
+
+    if self.builder_config.table_option == 'movies':
+      features_dict = {**movie_features_dict}
+    elif self.builder_config.format_version in ['1m', '100k']:
+      # Older versions of MovieLens have demographic features.
+      features_dict = {
+          **movie_features_dict,
+          **rating_features_dict,
+          **demographic_features_dict
+      }
+      if self.builder_config.format_version == '100k':
+        # Only the 100k dataset contains exact user ages. The 1m dataset
+        # contains only bucketized age values.
+        features_dict['raw_user_age'] = tf.float32
```

Comment:
I think 'flattening' the conditional logic though more verbose would be more readable, i.e.,

```suggestion
    elif self.builder_config.format_version == '1m':
      features_dict = {
          **movie_features_dict,
          **rating_features_dict,
          **demographic_features_dict,
      }
    elif self.builder_config.format_version == '100k':
      features_dict = {
          **movie_features_dict,
          **rating_features_dict,
          **demographic_features_dict,
          'raw_user_age': tf.float32,
      }
```

Reply:
This came across my mind a bit but I thought it might be a bit redundant. However it does improve readability a lot. The logic is flattened in ad8a410399cac16e254f4992ec7519ff46b14a16.